src/transactions/test/LiquidityPoolTradeTests.cpp

```diff
@@ -0,0 +1,322 @@
+// Copyright 2021 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "ledger/LedgerTxn.h"
+#include "ledger/LedgerTxnEntry.h"
+#include "ledger/LedgerTxnHeader.h"
+#include "lib/catch.hpp"
+#include "main/Application.h"
+#include "test/TestAccount.h"
+#include "test/TestExceptions.h"
+#include "test/TestUtils.h"
+#include "test/TxTests.h"
+#include "test/test.h"
+#include "transactions/TransactionUtils.h"
+
+using namespace stellar;
+using namespace stellar::txtest;
+
+TEST_CASE("liquidity pool trade", "[tx][liquiditypool]")
+{
+    VirtualClock clock;
+    auto app = createTestApplication(clock, getTestConfig());
+
+    // set up world
+    auto minBal = [&](int32_t n) {
+        return app->getLedgerManager().getLastMinBalance(n);
+    };
+    auto root = TestAccount::createRoot(*app);
+    auto native = makeNativeAsset();
+    auto cur1 = makeAsset(root, "CUR1");
+    auto cur2 = makeAsset(root, "CUR2");
+    auto share12 =
+        makeChangeTrustAssetPoolShare(cur1, cur2, LIQUIDITY_POOL_FEE_V18);
```

Comment:
note that `makeChangeTrustAssetPoolShare` doesn't enforce that `cur1 < cur2` - we should probably make it resilient to that (either by having a `REQUIRE(cur1 < cur2)` or by ordering internally like you did in `OfferExchange.cpp`)